tycho-extras/tycho-document-bundle-plugin/src/main/java/org/eclipse/tycho/extras/docbundle/JavadocRunner.java

```diff
@@ -33,294 +33,312 @@
+			count += addPackages(sb, bundle.getManifestElements("Export-Package"));
+		}
+		return count;
+	}
+
+	private int addPackages(final StringBuilder sb, final ManifestElement[] manifestElements) {
+		if (manifestElements == null) {
+			return 0;
+		}
+
+		for (final ManifestElement ele : manifestElements) {
+			final String pkg = ele.getValue();
+
+			final boolean include = includeMatcher != null ? includeMatcher.matches(pkg) : true;
+			final boolean exclude = excludeMatcher != null ? excludeMatcher.matches(pkg) : false;
+
+			if (include && !exclude) {
+				sb.append(pkg).append(lineSeparator);
+			}
+		}
+
+		return manifestElements.length;
+	}
+
+	private void addPath(final StringBuilder sb, final Collection<?> path) {
+		boolean first = true;
+		for (final Object ele : path) {
+			if (ele == null) {
+				continue;
+			}
+			// convert black slashes to forward slashes for javadoc
+			final String pathEle = ele.toString().replace('\\', '/');
+			if (!first) {
+				sb.append(File.pathSeparator);
+			} else {
+				first = false;
+			}
+			sb.append(pathEle);
+		}
+	}
+
+	private void addSourcePaths(final StringBuilder sb) {
+		addPathArgument(sb, "-sourcepath", this.sourceFolders);
+	}
+
+	private void addPathArgument(final StringBuilder sb, final String arg, final Collection<?> path) {
+		if (path.isEmpty()) {
+			return;
+		}
+		sb.append(arg);
+		sb.append(" '");
+		addPath(sb, path);
+		sb.append("'" + lineSeparator);
+	}
+
+	protected String getExecutable() {
+		log.debug("Find javadoc executable");
+
+		if (this.options.getExecutable() != null) {
+			// prefer the specific one
+			log.debug("Using specified javadoc: " + options.getExecutable());
+			return this.options.getExecutable();
+		}
+
+		log.debug("Toolchain manager: " + toolchainManager);
+
+		if (this.toolchainManager != null) {
+			// try the toolchain
+			final Toolchain tc = this.toolchainManager.getToolchainFromBuildContext("jdk", this.session);
+			log.debug("Toolchain: " + tc);
+
+			if (tc != null) {
+				final String exe = tc.findTool("javadoc");
+				log.debug("Toolchain Tool: " + exe);
+				if (exe != null) {
+					return exe;
+				}
+			}
+		}
+
+		String javaHome = System.getProperty("java.home");
+		String javadocFromJavaHome;
+
+		// derive path to javac from java.home similar to
+		// org.codehaus.plexus.compiler.javac.JavacCompiler.getJavacExecutable() in
+		// plexus-compiler-javac
+		if (OS.isFamilyMac()) {
+			javadocFromJavaHome = javaHome + File.separator + "bin" + File.separator + "javadoc";
+		} else {
+			javadocFromJavaHome = javaHome + File.separator + ".." + File.separator + "bin" + File.separator
+					+ "javadoc";
+		}
+
+		if (OS.isFamilyWindows()) {
+			javadocFromJavaHome += ".exe";
+		}
+
+		log.debug("Testing javadoc from java.home = " + javadocFromJavaHome);
+
+		if (new File(javadocFromJavaHome).canExecute()) {
+			return javadocFromJavaHome;
+		}
+
+		// use JAVA_HOME
```

Comment:
lines 303 to 317 are the actual change. Not sure what lead to showing everything as diff.

Reply:
never mind, I think this is due to changed formatting settings.